### PR TITLE
Add a webhook to add new users to a group

### LINKED ISSUE #101: Create a webhook to add new users to a group

### PR DESCRIPTION
**Pull request by ManujaDewmina**

### CODE REVIEW

File: webhook/webhook.bal [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        log:printInfo(event.toJsonString());
+    }
+    
+    remote function onAcceptUserInvite(asgardeo:GenericEvent event ) returns error? {
+        
+        log:printInfo(event.toJsonString());
+    }
+}
+
+service /ignore on httpListener {}
```
Comment (daneshk) [inline webhook/webhook.bal]:
```suggestion
service /ignore on httpListener {}

```

File: webhook/webhook.bal [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+scim:ConnectorConfig configscim = {
+    orgName: orgName,
+    clientId: clientId,
+    clientSecret : clientSecret,
+    scope : ["internal_group_mgt_update"]
+};
+
+listener http:Listener httpListener = new(8090);
+listener asgardeo:Listener webhookListener =  new(config,httpListener);
```
Comment (daneshk) [inline webhook/webhook.bal]: Can you format the code, please? there are few formatting issues

File: webhook/webhook.bal [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+service asgardeo:RegistrationService on webhookListener {
+
+    remote function onAddUser(asgardeo:AddUserEvent event ) returns error? {
+        json user = event.toJson();
+        json|error userName = user.eventData.userName;
+        json|error userId = user.eventData.userId;
```
Comment (daneshk) [inline webhook/webhook.bal:26]: without converting to JSON, shall we get the `username` and `userId` from the event record? like,
```
string userName = event.userName;
string userId = event.userId;
```

File: webhook/webhook.bal [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            log:printError("Error while getting userId", 'error = userId);
+            return userId;
+        } else {
+            string userNameStr = userName.toString();
+            string userIdStr = userId.toString();
+
+            scim:Client|error clientuser = new(configscim);
+            if clientuser is error {
+                log:printError("Error while creating scim client", 'error = clientuser);
+                return clientuser;
```
Comment (daneshk) [inline webhook/webhook.bal]: According to the webhook impl, when we return an error, it prints an error stack trace. so we don't need to print log here

File: webhook/webhook.bal (hunk 3 above)
Comment (ManujaDewmina) [reply webhook/webhook.bal:26]: It does not give access for members. 
`'ballerinax/trigger.asgardeo:0.5.0:AddUserData?' does not support field access'`

**Review by daneshk — APPROVED** [review]

**Pull request merged**